### PR TITLE
Module specifier

### PR DESCRIPTION
**Pull request by trotzig**
This addresses #288. I haven't added anything to the README because I wanted to first know if you think I'm on the right track. 

Also, we're currently at 1000 commits. We should just stop and call the project done. ;)

### CODE REVIEW

**Comment by trotzig** [comment]
Ping @rhettlivingston, @lencioni (I don't know if you subscribe to PR changes)

**Comment by lencioni** [comment]
Good work!

I'm not convinced about the name `moduleSpecifier`. It's not quite right. Maybe `moduleNameFormatter`?

Also, I'm not sure if there is an advantage to having the original `moduleSpecifier` passed as an argument to the user-defined one. While I like the explicitness of it, I wonder if it would be better to just fall back if the user-defined one returns `null` or `undefined`. However, if we expect to use this pattern in other places and it has advantages there, I can see the case for keeping APIs consistent.

File: lib/JsModule.js [hunk 1]
```diff
@@ -201,7 +201,10 @@ export default class JsModule {
       defaultImport,
       importFunction: config.get('importFunction', { pathToImportedModule }),
       namedImports,
-      path: this.importPath,
+      path: config.get('moduleSpecifier', {
+        pathToImportedModule,
+        moduleSpecifier: this.importPath,
+      }),
```
Comment (rhettlivingston) [inline lib/JsModule.js]: Is it OK that the "path" specified to ImportStatement has been changed from one that might have been useful for actually finding a file to one that is only useful as a ModuleSpecifier in an import statement? If so, should it be called "path"?

**Comment by rhettlivingston** [comment]
> Ping @rhettlivingston, @lencioni (I don't know if you subscribe to PR changes)

I've got a watch on the project for the foreseeable future, so I'm notified on everything.

File: lib/JsModule.js (hunk 1 above)
Comment (trotzig) [inline lib/JsModule.js]: Good point. I'll rename the variable internally for ImportStatement.

**Comment by trotzig** [comment]
Good, I'll stop the pinging then.

**Comment by trotzig** [comment]
@lencioni I like `moduleNameFormatter` for the option/function and `moduleName` for the variable. I'll change to that. We probably want the original `moduleName` passed in so that you can make tiny adjustments. I thought about falling back if the method returns undefined, but I think it's fine always having to return something.

File: lib/JsModule.js (hunk 1 above)
Comment (trotzig) [inline lib/JsModule.js]: That turned out to be non-trivial. We use `importPath` in the public API that the editor plugins use, and I'd rather push the rename to a different time as it requires updates to all the plugins as well. I could of course rename it locally only, but I'd rather keep a funky but consistent name than a more correct but inconsistent one.

Comment (lencioni) [inline lib/JsModule.js]: Maybe create an issue with a breaking change label so we don't forget about it?

Comment (trotzig) [inline lib/JsModule.js]: Good call. #311

**Comment by rhettlivingston** [comment]
This looks great to me!

File: README.md [hunk 2]
```diff
@@ -379,7 +380,33 @@ Defaults to `80`. This setting controls when import statements are broken into
...
+### `moduleNameFormatter`
+
+Use a function here to control how the resulting module name string will look
+like. It's useful if you for instance want to add a custom prefix to certain
+imports. Apart from the standard `pathToCurrentFile` and `pathToImportedModule`
+values passed in to all configuration functions, this method is also passed a
+`moduleName` value, which in general is what you want to manipulate.
+
+```js
+moduleNameFormatter: ({ moduleName, pathToCurrentFile } => {
```
Comment (rhettlivingston) [inline README.md]: This is missing a closing parentheses right before the arrow.

Comment (trotzig) [inline README.md]: Thanks! Fixed in d0d386a.

**Pull request merged**